impeller/aiks/picture.cc

```diff
@@ -3,11 +3,23 @@
 // found in the LICENSE file.
 
 #include "impeller/aiks/picture.h"
+#include <memory>
 
 #include "impeller/entity/entity.h"
+#include "impeller/renderer/render_target.h"
 
 namespace impeller {
 
-//
+std::shared_ptr<Image> Picture::RenderToImage(AiksContext& renderer) {
+  auto coverage = pass->GetElementsCoverage();
+  auto target = RenderTarget::CreateOffscreen(
+      *renderer.context_, ISize(coverage->GetRight(), coverage->GetBottom()));
```

Comment:
This will create an unnecessarily large texture if the coverage has an offset element. In that case, pass entities would need to be copied and offset applied (I believe we already do this in the entity pass).

Reply:
This is the behavior of the [`picture.toImage`](https://api.flutter.dev/flutter/dart-ui/Picture/toImage.html). I found it a tad less confusing to reason about as opposed to returning the position as part of the result or otherwise forcing the caller to fetch coverage from the picture separately. Up to you though -- I don't really have a strong opinion.

A good middle ground might be to add a convenience method for walking the picture and absorbing the top left coverage margin as you mention.

Reply:
In that case, I think we should return an `impeller::Snapshot` in //impeller/entity/contents. Returning the position as part of image is then taken care of. Introducing a potentially large allocation overhead to make the API easier to reason makes me nervous. Especially when we have followed the right pattern elsewhere and have utilities for this purpose already.

If you do decide to use the snapshot, perhaps just call the method `Picture::Snapshot` as well and also move `//impeller/entity/contents/snapshot` to someplace more easily discoverable. Perhaps `//impeller/renderer/snapshot`? I think it's a good way of thinking about snapshotting and you have already thought through the various ways in which snapshotting can be used.